pex/pip.py

```diff
@@ -170,14 +195,21 @@ def maybe_trust_insecure_host(url):
             yield "--proxy"
             yield network_configuration.proxy
 
-        if network_configuration.cert:
-            yield "--cert"
-            yield network_configuration.cert
+        # N.B.: `--cert` is passed via env var to work around:
+        #   https://github.com/pypa/pip/issues/5502
+        # See `_calculate_network_configuration_env`.
 
         if network_configuration.client_cert:
             yield "--client-cert"
```

Comment:
That pip issue mentions that the same problem exists for --client-cert, we might want to fix that similarly.

Reply:
Indeed. Looking at this in more detail, the current fix is thwarted by the fact that we run Pip in `--isolated` mode. That toggles off reading both pip configuration files (which you cannot customize the location of) and environment variables. Unfortunately there is no way to turn off just configuration files. Unlike the `--cert` case, which can be fixed by keeping `--isolated` and backdooring straight to requests using `REQUESTS_CA_BUNDLE` as @thamenato did, there is no such back door for the `--client-cert` case. I'll circle back with a change that switches to using the `REQUESTS_CA_BUNDLE` back door instead of `PIP_CERT` and then, iff `--client-cert` is specified, disable isolation, wrn about this, and use `PIP_CLIENT_CERT` to communicate that to subprocesses.

Reply:
I'm impressed with how this keeps coming back with deeper and deeper issues...